src/Components/Components/src/Rendering/RenderTreeBuilder.cs

```diff
@@ -505,6 +505,18 @@ public void OpenComponent(int sequence, [DynamicallyAccessedMembers(Component)]
         OpenComponentUnchecked(sequence, componentType);
     }
 
+    /// <summary>
+    /// Appends a frame representing a component parameter.
+    /// </summary>
+    /// <param name="sequence">An integer that represents the position of the instruction in the source code.</param>
+    /// <param name="name">The name of the attribute.</param>
+    /// <param name="value">The value of the attribute.</param>
+    public void AddComponentParameter(int sequence, string name, object? value)
```

Comment:
The one slightly unfortunate thing we lose by doing this is that `bool` parameter values now get boxed on a per-call basis. We could add a `bool` overload so we can use `BoxedTrue` and `BoxedFalse`, but that re-introduces part of the problem that this PR is solving (i.e., consider if a type had an implicit conversion to `bool`).

Reply:
If we really wanted to avoid this for `bool` we could have a further overload:

```cs
public void AddComponentParameter<T>(int sequence, string name, T value) where T : IEquatable<bool>
{
    if (typeof(T) == typeof(bool))
    {
        // ... use box cache
    }
    else
    {
        AddComponentParameter(sequence, name, (object)value);
    }
}
```

The `where T : IEquatable<bool>` constraint here is intended to reduce the number of cases where this gives a false-positive match (i.e., is selected for something unrelated to the scenario with bools) so we don't take the `if (typeof(T) == typeof(bool))` overhead in the vast majority of cases.

To be honest it seems unlikely that the `BoxedTrue`/`BoxedFalse` optimization is going to have a measurable impact for anyone. We only put it in because it was easy before. I would err on the side of *not* worrying about this impacting bools unless someone can produce numbers showing it matters in a plausibly real case. And then we always have the option of adding this special constrained generic overload in the future if we want.

Reply:
I agree, maybe we should wait until we see a need before trying to add this optimization back in.